### ISSUE TITLE
What does _, mean?

### ISSUE THREAD

**Issue by calvintwr** [Alamofire/Alamofire#654]
What does the _, in the code below mean?

Alamofire.request(.GET, "http://httpbin.org/get")
         .responseJSON { _, _, JSON, _ in
             println(JSON)
         }

**Comment by cristeahub** [comment]
You should ask these questions on stack overflow instead of here. Not only will you get a better response, but the person who responds will also get some points on the site, which is nice for them. See the guidelines here: https://github.com/Alamofire/Alamofire#communication .

You can read about how the underscore works here, it basically means that  you don't care about the value in those field: https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Declarations.html#//apple_ref/doc/uid/TP40014097-CH34-ID362

**Comment by calvintwr** [comment]
ahhh i see... thanks for your reply.

But, actually I would recommend that the documentation don't use _ in such a manner. Its sending out the wrong signal.

Because the status codes, which is inside the response, is important. And there should always be conditionals base on the status codes. People will read your documentation and get away with the idea that the response is not important. In fact I was stumped by this for quite some time...

**Comment by cristeahub** [comment]
I think it's like that because they want to emphasize the difference of a `responseJSON` and a regular `response`.

**Comment by cnoon** [comment]
Thank you for your input @calvintwr. We chose to use underscores to emphasize meaning and show intent just as @cristeahub mentioned. Since this is a VERY common pattern demonstrated all over the Swift language, we feel it is also a good pattern to use in the documentation in particular areas. If you are unfamiliar with the pattern, then I suggest you read up on Swift closures and usage patterns to get a better feel for what we're trying to demonstrate. Again, thanks for the feedback.

Cheers.

**Comment by calvintwr** [comment]
Hi cnoon,

Thanks for the reply. Yup I fully understand the intent now.

But my concern is just that users of the documentation are not always that familiar with swift, and this will trip them.

As an example, my colleague has 5 years of experience coding ObjectiveC and just recently attempted to switch to swift.

I have even more years doing many other languages.

But we didn't understand the documentation in this respect, and I certainly don't expect new users of swift to be able to pour through the swift docs.

In short, I don't think it's wise to craft documentation meant for adept level of readers, certainly not for a framework that is so high level and simple.

Cheers mate.

**Comment by cnoon** [comment]
Hi @calvintwr, thanks for the feedback. I definitely see your point, but respectfully disagree. We're not catering the documentation to users that are not familiar with Swift. I'd also discourage users from attempting to use Alamofire before getting familiar with all the ins and outs of the Swift language.
